modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/to/SearchHistoryEntry.kt

```diff
@@ -0,0 +1,133 @@
+        val term: String,
+    ) : SearchHistoryEntry(id = id)
+
+    fun toSearchHistoryItem() = when (this) {
+        is Episode -> SearchHistoryItem(
+            id = id,
+            episode = SearchHistoryItem.Episode(
+                uuid = uuid,
+                title = title,
+                publishedDate = publishedDate,
+                duration = duration
+            )
+        )
+
+        is Folder -> SearchHistoryItem(
+            id = id,
+            folder = SearchHistoryItem.Folder(
+                uuid = uuid,
+                title = title,
+                color = color,
+                podcastIds = podcastIds.joinToString()
+            )
+        )
+
+        is Podcast -> SearchHistoryItem(
+            id = id,
+            podcast = SearchHistoryItem.Podcast(
+                uuid = uuid,
+                title = title,
+                author = author
+            )
+        )
+
+        is SearchTerm -> SearchHistoryItem(
+            id = id,
+            term = term
+        )
+    }
+
+    companion object {
+        fun fromEpisode(episode: EpisodeModel) = Episode(
+            uuid = episode.uuid,
+            title = episode.title,
+            publishedDate = episode.publishedDate,
+            duration = episode.duration
+        )
+
+        fun fromFolder(folder: FolderModel, podcastIds: List<String>) = Folder(
+            uuid = folder.uuid,
+            title = folder.name,
+            color = folder.color,
+            podcastIds = podcastIds
+        )
+
+        fun fromPodcast(podcast: PodcastModel) = Podcast(
+            uuid = podcast.uuid,
+            title = podcast.title,
+            author = podcast.author
+        )
+
+        fun fromSearchHistoryItem(item: SearchHistoryItem) = when {
+            item.episode != null -> {
+                val episode = item.episode as SearchHistoryItem.Episode
+                Episode(
+                    id = item.id,
+                    uuid = episode.uuid,
+                    title = episode.title,
+                    publishedDate = episode.publishedDate,
+                    duration = episode.duration
+                )
+            }
+
+            item.folder != null -> {
+                val folder = item.folder as SearchHistoryItem.Folder
+                Folder(
+                    id = item.id,
+                    uuid = folder.uuid,
+                    title = folder.title,
+                    color = folder.color,
+                    podcastIds = folder.podcastIds.split(",")
+                )
+            }
+
+            item.podcast != null -> {
+                val podcast = item.podcast as SearchHistoryItem.Podcast
+                Podcast(
+                    id = item.id,
+                    uuid = podcast.uuid,
+                    title = podcast.title,
+                    author = podcast.author
+                )
+            }
+
+            else -> SearchTerm(id = item.id, term = item.term ?: "")
```

Comment:
For search terms, should the `item.term` never be `null`? If so, do you think it's worth checking that specifically here (instead of using `else ->`) and instead use the `else ->` branch to log an error message or throw an exception so it'll be easier to catch it if a bug gets introduced here in the future?

Reply:
Good idea, added in f6f2367